### PR TITLE
11.0 subscription mass mailing

### PR DESCRIPTION
**Pull request by houssine78**
https://gestion.coopiteasy.be/web#id=3913&view_type=form&model=project.task&menu_id=338&action=479

### CODE REVIEW

File: distribution_circuits_base/demo/demo.xml [hunk 1]
```diff
@@ -5,7 +5,7 @@
  License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
 -->
 <odoo>
-    <data noupdate="0"> <!-- fixme -->
+    <data noupdate="1"> <!-- fixme -->
```
Comment (robinkeunen) [inline distribution_circuits_base/demo/demo.xml]: You can remove the fixme tag

File: distribution_circuits_base/demo/demo.xml [hunk 2]
```diff
@@ -71,7 +71,7 @@
             <field name="country_id" ref="base.be"/>
         </record>
 
-        <record id="partner_gac_1" model="res.partner">
+        <record id="distribution_circuits_base.partner_gac_1" model="res.partner">
```
Comment (robinkeunen) [inline distribution_circuits_base/demo/demo.xml]: You're already in that module, why did you need to add the prefix? Why did you need to change this?

File: subscription_mass_mailing/models/mail.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+from odoo import models, fields
+
+
+class Mail(models.Model):
+    _inherit = "mail.mail"
+
+#     subscription_id = fields.Many2one(
+#         comodel_name="subscription", string="Subscription"
+#     )
```
Comment (robinkeunen) [inline subscription_mass_mailing/models/mail.py]: remove if uneeded

File: subscription_mass_mailing/models/mailing_criterium.py [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+    )
+    mail_template = fields.Many2one(
+        comodel_name="mail.template",
+        string="Email Template",
+        required=True,
+    )
+    py_expr_filter = fields.Char(
+        string="Python Expression Filter",
+        required=False,
+        help="Available variables:\n"
```
Comment (robinkeunen) [inline subscription_mass_mailing/models/mailing_criterium.py:56]: Trailing `\n` would be unnecessary if you used triple-quoted comments.

File: subscription_mass_mailing/models/mailing_criterium.py [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+    @api.multi
+    def send_email(self):
+        for criterium in self:
+            subscribers = criterium.get_subscribers()
+            _logger.info(
+                'Generating %s emails for criterium "%s"'
+                % (len(subscribers), criterium.name)
+            )
+            for subscriber in subscribers:
+                criterium.mail_template.use_default_to = True
```
Comment (robinkeunen) [inline subscription_mass_mailing/models/mailing_criterium.py:94]: What does `use_default_to` do?

File: subscription_mass_mailing/models/subscription.py [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import models, api
+
+
+class Subscription(models.Model):
+    _inherit = "subscription"
+
+    @api.multi
+    def message_get_default_recipients(self):
```
Comment (robinkeunen) [inline subscription_mass_mailing/models/subscription.py]: I don't see the function  being used in the code.

**Review by robinkeunen — APPROVED** [review]
Small comments and questions, otherwise LGTM 👍

File: distribution_circuits_base/demo/demo.xml (hunk 2 above)
Comment (houssine78) [reply distribution_circuits_base/demo/demo.xml]: in fact this was for solving an issue because the demo file was in data and not in demo. I just forget to remove this.

File: subscription_mass_mailing/models/mailing_criterium.py (hunk 4 above)
Comment (houssine78) [reply subscription_mass_mailing/models/mailing_criterium.py:56]: I toke it like it was it the product_subscription_mass_mailing. I won't change it for now.

File: subscription_mass_mailing/models/mailing_criterium.py (hunk 5 above)
Comment (houssine78) [reply subscription_mass_mailing/models/mailing_criterium.py:94]: I don't dude this is your code :D

File: subscription_mass_mailing/models/subscription.py (hunk 6 above)
Comment (houssine78) [reply subscription_mass_mailing/models/subscription.py]: ok I removed it

File: distribution_circuits_base/demo/demo.xml (hunk 1 above)
Comment (houssine78) [reply distribution_circuits_base/demo/demo.xml]: indeed !

File: subscription_mass_mailing/models/mail.py (hunk 3 above)
Comment (houssine78) [reply subscription_mass_mailing/models/mail.py]: done

File: subscription_mass_mailing/models/mailing_criterium.py (hunk 5 above)
Comment (robinkeunen) [reply subscription_mass_mailing/models/mailing_criterium.py:94]: haha, I guess I copied that from somewhere too :p
